### PR TITLE
[docs] Do not list explicit versions / imagetags in README

### PR DESCRIPTION
**Pull request by gyfora**
## What is the purpose of the change

Remove explicit list of versions/image tags from README . These versions are already listed/updated on the Flink website during the upgrade process.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,15 +26,10 @@ For the complete feature-set please refer to our [documentation](https://nightli
...
-## Version Matrix
+To download the latest stable version please visit the [Flink Downloads Page](https://flink.apache.org/downloads.html).
+The official operator images are also available on [Dockerhub](https://hub.docker.com/r/apache/flink-kubernetes-operator/tags).
 
-| Operator Version     | API Version | Flink Compatibility | Operator Image                           |
-|----------------------|-------------|---------------------|------------------------------------------|
-| `latest` (main HEAD) | `v1beta1`   | 1.13, 1.14, 1.15    |                                          |
-| `release-1.0.1`      | `v1beta1`   | 1.13, 1.14, 1.15    | `apache/flink-kubernetes-operator:1.0.1` |
-| `release-0.1.0`      | `v1alpha1`  | 1.14                | `apache/flink-kubernetes-operator:0.1.0` |
+Please check out our docs to read about the [upgrade process](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/operations/upgrade/) and our [backward compatibility guarantees](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/operations/compatibility/).
```
Comment (wangyang0918) [inline README.md:32]: If we remove this version matrix, where the users could find the compatible Flink versions with specific operator release?

Comment (gyfora) [reply README.md:32]: The downloads page already shows this information. The README is only a pain in the butt during the realease process as you have to update it in addition to updating everything else:

image

Comment (wangyang0918) [reply README.md:32]: Make sense.

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Pull request merged**